tst/EnergyPlus/unit/ChillerElectricEIR.unit.cc

```diff
@@ -563,13 +563,46 @@ TEST_F(EnergyPlusFixture, ChillerElectricEIR_WaterCooledChillerVariableSpeedCond
     Real64 ActualCondFlow = 3.0 * std::abs(MyLoad) / (Cp * 10.0);
     EXPECT_NEAR(thisChiller.CondMassFlowRate, ActualCondFlow, 0.00001);
 
-    thisChiller.CondenserFlowControl = DataPlant::CondenserFlowControl::ConstantFlow;
-    thisChiller.calculate(*state, MyLoad, RunFlag);
-    EXPECT_NEAR(thisChiller.CondMassFlowRate, thisChiller.CondMassFlowRateMax, 0.00001);
-
     // Test the minimum condenser flow rate
     MyLoad = -500;
     thisChiller.CondenserFlowControl = DataPlant::CondenserFlowControl::ModulatedChillerPLR;
     thisChiller.calculate(*state, MyLoad, RunFlag);
     EXPECT_NEAR(thisChiller.CondMassFlowRate, thisChiller.CondMassFlowRateMax * 0.35, 0.00001);
+
+    // Test constant flow condenser
+    thisChiller.CondenserFlowControl = DataPlant::CondenserFlowControl::ConstantFlow;
+    MyLoad = -10000;
+    Real64 savedMyLoad = MyLoad;
+
+    // test with condenser flow available
+    thisChiller.initialize(*state, RunFlag, MyLoad);
+    thisChiller.calculate(*state, MyLoad, RunFlag);
+    thisChiller.update(*state, MyLoad, RunFlag);
+    Real64 chWOutletTemp = thisChiller.EvapInletTemp + savedMyLoad / (Cp * thisChiller.EvapMassFlowRate);
+    Real64 condOutletTemp = thisChiller.CondInletTemp + thisChiller.QCondenser / (Cp * thisChiller.CondMassFlowRate);
+    EXPECT_EQ(MyLoad, savedMyLoad);
+    EXPECT_NEAR(thisChiller.CondMassFlowRate, thisChiller.CondMassFlowRateMax, 0.00001);
+    EXPECT_NEAR(thisChiller.EvapMassFlowRate, thisChiller.EvapMassFlowRateMax, 0.00001);
+    EXPECT_NEAR(thisChiller.EvapOutletTemp, chWOutletTemp, 0.1);
+    EXPECT_NEAR(thisChiller.CondOutletTemp, condOutletTemp, 0.1);
+    EXPECT_NEAR(thisChiller.QEvaporator, -savedMyLoad, 1.0);
+    EXPECT_NEAR(thisChiller.QCondenser, (-savedMyLoad + thisChiller.Power), 1.0);
+    EXPECT_NEAR(thisChiller.Power, 20987, 1.0);
+
+    // test with no condenser flow available - chiller should be off
+    state->dataLoopNodes->Node(thisChiller.CondInletNodeNum).MassFlowRate = 0.0;
+    state->dataLoopNodes->Node(thisChiller.CondInletNodeNum).MassFlowRateMaxAvail = 0.0;
```

Comment:
MaxAvail does need to be set to 0 here, at the inlet node, but I wonder why. It's a limit. If the chiller requests 0 then it should set to 0 if not in series with another component? FlowCtrl = Invalid in this unit test so it's not series active. I tested this unit test both ways, with this line commented out and with the MaxAvail = 1. It failed both times. I guess the plant manager will set MaxAvail as necessary, at the condenser inlet, and this line is needed for that reason. The chiller can be off and still have flow at the condenser and/or evaporator if the plant requests it (e.g., no bypass loop).